elearning-frontend/src/pages/UserSinglePage.js

```diff
@@ -0,0 +1,88 @@
+import React, { useEffect, useMemo, useState } from 'react'
+import { useParams } from 'react-router-dom'
+import UserAvatar from '../components/UserAvatar'
+import Button from '../core/Button'
+import Toastify from '../core/Toastify'
+import apiClient from '../services/api'
+
+const UserSinglePage = () => {
+    let { id } = useParams()
+
+    const [user, setUser] = useState(null)
+    const [loading, setLoading] = useState(true)
+    
+    const getUser = (async () => {
+        const data = await apiClient({
+            method: 'get',
+            url: `/api/user/${id}`
+        }).then(response => {
+            setUser({...response.data})
+            setLoading(false)
+        }).catch(error => {
+            Toastify('error', error)
+        })
+    })
+
+    useEffect(() => {
+        getUser()
+    }, [])
+
+    const followUser = (e) => {
+        e.preventDefault()
+
+        apiClient({
+            method: 'post',
+            url: `/api/user/${user.id}/follow`,
+        }).then(response => {
+            Toastify('success', `You are now a follower of ${user.profile.firstname}`)
+        }).catch(error => {
+            Toastify("error", error)
+        })
+    }
+
+    const unfollowUser = (e) => {
+        e.preventDefault()
+
+        apiClient({
+            method: 'delete',
+            url: `/api/user/${user.id}/unfollow`,
+        }).then(response => {
+            Toastify('success', `You are now a follower of ${user.profile.firstname}`)
+        }).catch(error => {
+            Toastify("error", error)
+        })
+    }
+
+    var view_element = ''
+
+    if (loading) {
+        view_element = <p>Loading data</p>
+    }
+    else {
+        view_element =
+            <>
+                <UserAvatar avatar={ user.profile.avatar } />
+                <small className='mb-2'>{ `${user.profile.firstname} ${(user.profile.middlename === '' ? '' : user.profile.middlename + ' ')}${user.profile.lastname}` }</small>
+                <div className='flex flex-row items-center gap-2'>
+                    <Button
+                        text='FOLLOW'
+                        type='button'
+                        classes='bg-blue-500 hover:bg-blue-700'
+                        onClick={e => followUser(e) } />
+                    <Button
+                        text='UNFOLLOW'
+                        type='button'
+                        classes='bg-red-500 hover:bg-red-700'
+                        onClick={e => unfollowUser(e) } />
+                </div>
+            </>
+    }
```

Comment:
You can just transfer it below. no need to create a variable and then call it.
It is more cleaner